### PR TITLE
Make TPE import data more robust

### PR DESCRIPTION
**Pull request by liuzhe-lz**
### Description ###

The exported data format and/or resuming metrics format has changed sometime before and they are mismatch now.

I really don't want to investigate into those messy stuffs. So make TPE tolerate any metrics format.

It seems this problem applies to other tuners as well. But I don't want to touch them before a systematic refactor.

### Checklist ###
  - ~~test case~~
  - ~~doc~~

### How to test ###

Run an HPO experiment. Export then import.

### CODE REVIEW

File: nni/algorithms/hpo/tpe_tuner.py [hunk 1]
```diff
@@ -201,9 +202,15 @@ def trial_end(self, parameter_id, _success, **kwargs):
...
+            data = nni.load(data)
         for trial in data:
+            if isinstance(trial, str):
+                trial = nni.load(trial)
             param = format_parameters(trial['parameter'], self.space)
             loss = trial['value']
+            if isinstance(loss, dict) and 'default' in loss:
+                loss = loss['default']
             if self.optimize_mode is OptimizeMode.Maximize:
                 loss = -trial['value']
```
Comment (QuanluZhang) [inline nni/algorithms/hpo/tpe_tuner.py:215]: loss = -loss

**Review by ultmaster — APPROVED** [review]

**Comment by QuanluZhang** [comment]
better to pick the logic of data format out of `import_data`, leave to the next pr
